internal/app/machined/pkg/controllers/block/user_volume_config.go

```diff
@@ -364,6 +367,49 @@ func (ctrl *UserVolumeConfigController) handlePartitionUserVolumeConfig(
 	return nil
 }
 
+func (ctrl *UserVolumeConfigController) handleDiskUserVolumeConfig(
+	userVolumeConfig configconfig.UserVolumeConfig,
+	v *block.VolumeConfig,
+	volumeID string,
+) error {
+	diskSelector, ok := userVolumeConfig.Provisioning().DiskSelector().Get()
+	if !ok {
+		// this shouldn't happen due to validation
+		return fmt.Errorf("disk selector not found for volume %q", volumeID)
+	}
+
+	v.TypedSpec().Type = block.VolumeTypeDisk
```

Comment:
one thought - should we fail in the `VolumeManager` controller if the disk selector spec for VolumeTypeDisk  matches more than a single disk? as this might lead to undefined behavior e.g. on reboot?